src/libs/portfolio/portfolio.ts

```diff
@@ -264,16 +273,40 @@ export class Portfolio {
             // this is what to look for in the coingecko response object
             responseIdentifier: geckoResponseIdentifier(token.address, this.network)
           })
-          const priceIn: Price[] = Object.entries(priceData || {}).map(([baseCurr, price]) => ({
+          priceIn = Object.entries(priceData || {}).map(([baseCurr, price]) => ({
             baseCurrency: baseCurr,
             price: price as number
           }))
           if (priceIn.length) priceCache.set(token.address, [Date.now(), priceIn])
-          /* eslint-disable-next-line no-param-reassign */
-          token.priceIn = priceIn
-        } catch {
-          /* eslint-disable-next-line no-param-reassign */
-          token.priceIn = []
+        } catch (error: any) {
+          const errorMessage = error?.message || 'Unknown error'
+          priceIn = []
+
+          // Avoid duplicate errors, because this.bachedGecko is called for each token and if
+          // there is an error it will most likely be the same for all tokens
+          if (
+            !errors.find(
+              (x) =>
+                x.name === PORTFOLIO_LIB_ERROR_NAMES.PriceFetchError && x.message === errorMessage
+            )
+          ) {
+            errors.push({
+              name: PORTFOLIO_LIB_ERROR_NAMES.PriceFetchError,
+              message: errorMessage
+            })
+          }
+        }
+
+        if (!priceIn.length) {
+          tokensWithPriceErrors.push({
+            error: `Failed to fetch price data from cena.ambire.com for ${token.symbol} on ${networkId}`,
+            address: token.address
+          })
+        }
```

Comment:
This may be redundant, because the only scenario in which we store such error is when the token doesn't exist in coingecko, which would be obvious if `priceIn` is empty and there is no error for the whole network.